msal/tests/Test.MSAL.NET.Unit.net45/CacheV2Tests/InMemoryCachePathStorageTests.cs

```diff
@@ -0,0 +1,389 @@
+        // not get corrupted
+        [TestMethod]
+        public void WriteFileConcurrently()
+        {
+            Debug.WriteLine("TEST START: WriteFileConcurrently");
+
+            ThreadTestUtils.ParallelExecute(
+                () =>
+                {
+                    Thread.Sleep(new Random().Next() % 10);
+                    _io.Write(FileName, _data);
+                });
+
+            CollectionAssert.AreEqual(_data, _io.Read(FileName));
+
+            // The file is in the right location
+            Assert.IsTrue(_io.RootDirectory.FileExists(FileName));
+        }
+
+        // Tests file creation and deletion
+        [TestMethod]
+        public void TestDeleteFile()
+        {
+            Debug.WriteLine("TEST START: TestDeleteFile");
+
+            _io.Write(FileName, _data);
+            Assert.IsTrue(_io.RootDirectory.FileExists(FileName));
+
+            _io.DeleteFile(FileName);
+            Assert.IsFalse(_io.RootDirectory.FileExists(FileName));
+        }
+
+        // Tests FileIO::DeleteFile() and FileIO::Write() concurrently
+        [TestMethod]
+        public void WriteDeleteConcurrently()
+        {
+            Debug.WriteLine("TEST START: WriteDeleteConcurrently");
+
+            var actions = new List<Action>();
+            for (int i = 0; i < 100; i++)
+            {
+                actions.Add(
+                    () =>
+                    {
+                        Thread.Sleep(new Random().Next() % 10);
+                        _io.Write(FileName, _data);
+                    });
+                actions.Add(
+                    () =>
+                    {
+                        Thread.Sleep(new Random().Next() % 10);
+                        _io.DeleteFile(FileName);
+                    });
+            }
+
+            Parallel.Invoke(
+                new ParallelOptions
+                {
+                    MaxDegreeOfParallelism = 8
+                },
+                actions.ToArray());
+
+            // Either the file is in a valid state or doesn't exist
+            if (_io.RootDirectory.FileExists(FileName))
+            {
+                CollectionAssert.AreEqual(_data, _io.Read(FileName));
+            }
+        }
+
+        // Tests FileIO::Read(), FileIO::Write(), and FileIO::DeleteFile() concurrently
+        [TestMethod]
+        public void ReadWriteDeleteConcurrently()
+        {
+            Debug.WriteLine("TEST START: ReadWriteDeleteConcurrently");
+
+            int successfulReadsCount = 0;
+
+            var actions = new List<Action>();
+            for (int i = 0; i < 100; i++)
+            {
+                actions.Add(
+                    () =>
+                    {
+                        Thread.Sleep(new Random().Next() % 10);
+                        byte[] actualData = _io.Read(FileName);
+
+                        // Either the file doesn't exist
```

Comment:
>Either [](start = 27, length = 6)

the remaining part of the sentence is missing. 
Either the file doesn't exist or the file is empty ?

Reply:
Fixed and cleaned up